server/text_generation_server/utils/tokens.py

```diff
@@ -57,7 +60,13 @@ def __init__(
             if grammar != ""
             else None
         )
+        self.logit_bias_processor = (
+            LogitBiasProcessor(logit_bias, tokenizer, device)
+            if logit_bias is not None and len(logit_bias) > 0
+            else None
+        )
         self.tokenizer = tokenizer
+        self.logit_bias = logit_bias
```

Comment:
This is not necessary, once we have the processor we should let go of the other object (Think of this as logit_bias taking ownership)

Reply:
oo yea removed 🙏